lms/djangoapps/courseware/tests/test_masquerade.py

```diff
@@ -374,6 +375,51 @@ def test_masquerade_as_specific_student(self):
         self.login_student()
         self.assertEqual(self.get_progress_detail(), u'2/2')
 
+    @patch.dict('django.conf.settings.FEATURES', {
+        'DISABLE_START_DATES': False,
+        'ENABLE_UNICODE_USERNAME': True
+    })
+    def test_masquerade_as_specific_student_with_special_char_uname(self):
```

Comment:
This is mostly a duplicate of `test_masquerade_as_specific_student`. Please use `@ddt` whenever possible:

```
@patch.dict('django.conf.settings.FEATURES', {'DISABLE_START_DATES': False})
@ddt.data('john', 'foo@bar')
def test_masquerade_as_specific_student(self, username):
```

You'd also need to add the lines on `test_masquerade_as_specific_student`
```
    student = UserFactory.create(username=username)
    CourseEnrollment.enroll(student, self.course.id)
    self.logout()
    self.login(student.email, 'test')
```

Wrapping the class with `@ddt.ddt` is also needed:
```
@ddt.ddt
TestStaffMasqueradeAsSpecificStudent
```